ShippingApp/src/main/java/com/javaschool/queue/QueueSender.java

```diff
@@ -0,0 +1,108 @@
+package com.javaschool.queue;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.javaschool.common.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.amqp.core.AmqpTemplate;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class QueueSender {
+
+    private static final Logger logger = LoggerFactory.getLogger(QueueSender.class);
+
+    private AmqpTemplate rabbitTemplate;
+
+    private GlobalProperties globalProperties;
+
+    private ObjectMapper objectMapper;
+
+    public QueueSender(final AmqpTemplate rabbitTemplate, GlobalProperties globalProperties) {
+        this.rabbitTemplate = rabbitTemplate;
+        this.globalProperties = globalProperties;
+        this.objectMapper = new ObjectMapper();
+    }
+
+    private String sendMessage(String queueMessage) {
+        String message = (String) rabbitTemplate.convertSendAndReceive(globalProperties.getExchangeShipping(), globalProperties.getRoutingKeyShipping(), queueMessage);
+        logger.info(message);
+        return message;
+    }
+
+    private String messageRequest(String type){
+        QueueMessageRequest messageRequest = new QueueMessageRequest();
+        String message;
+        String queueResponse = "";
+        messageRequest.setType(type);
+        try {
+            message = objectMapper.writeValueAsString(messageRequest);
+            queueResponse = sendMessage(message);
+        } catch (JsonProcessingException e) {
+            logger.error(e.getMessage());
+        }
+        return queueResponse;
+    }
+
+    public List<String> getSize() {
```

Comment:
There's a lot of repeated code within `getSize`, `getType` and `getTime` methods. It should be a better way.

Reply:
Check Java generics to remove boilerplate code